### PR TITLE
Added jammed endstop detection

### PR DESCRIPTION
**Pull request by Treysen-Zobell**
Throws error if relevant endstop was already triggered for XYZ homing. Detection not added to U axis because both natural states leave it triggered.

### CODE REVIEW

File: software/duet_config_files/duet2/rrf2/macros/config_checks.g [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+; Z
+echo "Please depress Z probe for 1s"
+while sensors.probes[0].value[0] == 0
+  G4 P500
+echo "Z probe test passed"
+
+; U
+echo "Rotate U axis to minimum"
+M577 U S1
+echo "Rotate U Axis to maximum (*insert spring stretch explanation)"
```
Comment (Poofjunior) [inline software/duet_config_files/duet2/rrf2/macros/config_checks.g:38]: This gave me a good chuckle. I can fill this in later.

File: software/duet_config_files/duet2/rrf2/macros/config_checks.g [hunk 2]
```diff
@@ -0,0 +1,73 @@
+; Machine Prechecks
+M84
+M291 R"Machine Prechecks" P"Please move printhead to center of bed, and bed to middle height, and tool changer axis to middle." S3
```
Comment (Poofjunior) [inline software/duet_config_files/duet2/rrf2/macros/config_checks.g:3]: I would make a note that they can do this now without turning off the machine. Something like:

"Motors have been disabled and can now be rotated by hand. <Rest of original message>"

Let's also refer to the printhead as the carriage if it doesn't have tools.

File: software/duet_config_files/duet2/rrf2/macros/config_checks.g [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+echo "Y endstop test passed"
+
+; Z
+echo "Please depress Z probe for 1s"
+while sensors.probes[0].value[0] == 0
+  G4 P500
+echo "Z probe test passed"
+
+; U
+echo "Rotate U axis to minimum"
```
Comment (Poofjunior) [inline software/duet_config_files/duet2/rrf2/macros/config_checks.g:36]: Maybe we could be be more explicit about what minimum means. Something like:
"Rotate U axis until it clicks the external switch. (minimum travel)"

File: software/duet_config_files/duet2/rrf2/macros/config_checks.g (hunk 1 above)
Comment (Poofjunior) [reply software/duet_config_files/duet2/rrf2/macros/config_checks.g:38]: In the meantime, we can probably just say:
"Rotate U axis in the opposite direction until you hear it click the internal switch. (maximum travel)"

File: software/duet_config_files/duet2/rrf3/macros/first_run.g [hunk 4]
```diff
@@ -0,0 +1,73 @@
+; Machine Prechecks
+M84
+M291 R"Machine Prechecks" P"Please move printhead to center of bed, and bed to middle height, and tool changer axis to middle." S3
```
Comment (Poofjunior) [inline software/duet_config_files/duet2/rrf3/macros/first_run.g:3]: Same as above:
I would make a note that they can do this now without turning off the machine. Something like:

"Motors have been disabled, and machine axes can now be adjusted by hand. [Rest of original message]"

Let's also refer to the printhead as the carriage if it doesn't have tools.

**Comment by Poofjunior** [comment]
Overall, this looks good.
It looks like config_checks.g and first_run.g are... the same file? Let's keep just one if that's the case.

If you can run through and tweak all the items above, I'll run through a hardware check on my end and get it merged. Thanks again for the contribution and for being patient on getting it merged!

**Comment by Treysen-Zobell** [comment]
I'll still make those changes, but I didn't mean to include the first run checks, only homex.g, homey.g, and homez.g. The first run checks are definitely incomplete.

**Comment by Poofjunior** [comment]
Oh, hey, in that case, we can do one of two things.

If you pop those files out of that branch and into another branch, they'll just disappear from this PR. (Can work you through the details if you need em.)

Or I can make a **dev** branch and you can reopen this PR into that branch. We can then merge as-is, knowing the **dev** branch has work to be done. And I'm happy to keep developing the parts I noted above if you do that too.

All-in-all, a bringup self-check script is a huge win for a ton of first-time builders--even if it's minus a few features.

Let me know how you'd like to proceed--and cheers!

**Pull request closed without merging**